### PR TITLE
Run coverage on pull requests

### PR DESCRIPTION
**Pull request by oxve**
b/286924777

### CODE REVIEW

File: .github/workflows/linux.yaml
Comment (isarkis) [inline .github/workflows/linux.yaml:60]: This will result in coverage running on nightlies and all branches, is that intended?

**Review by isarkis — COMMENTED** [review]
This is awesome!  I was wondering how user can know coverage % for their particular PR.  The comment shows diff between baseline and after PR is merged.

File: .github/workflows/linux.yaml
Comment (isarkis) [reply .github/workflows/linux.yaml:60]: Actually, I take it back.  It will run on nightlies, but not all branches so long as this code is in main it will run on main and feature/* branches.

**Review by arjungm — COMMENTED** [review]
should we add a tag like "skip-coverage" for stuff that shouldn't be tested for coverage? I could see this being a blocker for branch-specific work for releases

File: .github/workflows/linux.yaml
Comment (oxve) [reply .github/workflows/linux.yaml:60]: I don't think nightlies make sense. If we already run on push we have continuous tracking.

I'll update.

**Comment by oxve** [comment]
> This is awesome! I was wondering how user can know coverage % for their particular PR. The comment shows diff between baseline and after PR is merged.

I haven't seen this tested with changes that include actual source code with tests. I think the comment will include coverage info about the files modified. I hope it's cool!


> should we add a tag like "skip-coverage" for stuff that shouldn't be tested for coverage? I could see this being a blocker for branch-specific work for releases

Since we're not blocking on this check (yet) I don't think we need to add any labels to avoid it (yet).

**Review by arjungm — APPROVED** [review]

File: .github/workflows/linux.yaml
Comment (oxve) [reply .github/workflows/linux.yaml:60]: Ok, reverted to the original check and added the `pull_request` event

File: .github/workflows/linux.yaml
Comment (isarkis) [inline .github/workflows/linux.yaml:60]: Any reason we don't want to run coverage on the feature/* branches?  Also, I think we want to run this on workflow_dispatch event.

Comment (oxve) [reply .github/workflows/linux.yaml:60]: I don't think so. If you want to see coverage you can either run it locally or create a PR. Two very valid workarounds.

Comment (oxve) [reply .github/workflows/linux.yaml]: > Also, I think we want to run this on workflow_dispatch event.

Done

**Pull request merged**
